app/src/main/java/org/wikipedia/gallery/GalleryItemFragment.java

```diff
@@ -209,6 +209,9 @@ public boolean onOptionsItemSelected(MenuItem item) {
      * @param pagerPosition    The pager's current position that is displayed to the user.
      */
     public void onUpdatePosition(int fragmentPosition, int pagerPosition) {
+        if (!isAdded()) {
```

Comment:
This might fix this particular crash, but it doesn't address the root of the problem, which is that in `GalleryActivity`, the `imageCaptionDisposable` never gets disposed.  We should probably dispose it in `onDestroy`.

Reply:
Makes sense and done.

I added the `imageCaptionDisposable.dispose();`, please let me know if you think this logic of checking  `isAdded()` should be removed.